x/capability/types/genesis.go

```diff
@@ -13,27 +14,45 @@ type GenesisState struct {
 	// capability global index
 	Index uint64 `json:"index" yaml:"index"`
 
-	Owners []Owner `json:"owners" yaml:"owners"`
+	// map from index to owners of the capability index
+	// index key is string to allow amino marshalling
+	Owners map[string]CapabilityOwners `json:"owners" yaml:"owners"`
```

Comment:
Key needed to be a `string` instead of `uint64` or amino JSONMarshal method paniced. Native go json-marshalling accepts uint64 keys in maps. Will ditching amino for good allow us to use uint64 keys instead?

Reply:
It’s prob the mapping that is causing the non determinism

Reply:
Ahh that makes sense

Reply:
Do not use a map! Instead, create a concrete structure that contains the capability owners and the respective index.

```diff
diff --git a/x/capability/types/genesis.go b/x/capability/types/genesis.go
index 85c7f4162..72964a83a 100644
--- a/x/capability/types/genesis.go
+++ b/x/capability/types/genesis.go
@@ -2,10 +2,16 @@ package types
 
 // GenesisState represents the Capability module genesis state
 type GenesisState struct {
-       Index uint64 `json:"index" yaml:"index"`
+       Index  uint64 `json:"index" yaml:"index"`
+       Owners []GenesisCapabilityOwners
 }
 
 // DefaultGenesis returns the default Capability genesis state
 func DefaultGenesis() GenesisState {
        return GenesisState{Index: 1}
 }
+
+type GenesisCapabilityOwners struct {
+       Index  uint64
+       Owners CapabilityOwners
+}
```